src/main/kotlin/uk/gov/justice/digital/hmpps/prisonercontactregistry/client/PrisonApiClient.kt

```diff
@@ -58,10 +59,21 @@ class PrisonApiClient(
   }
 
   fun getPersonAddress(personId: Long): List<AddressDto>? {
+    val uri = "/api/persons/$personId/addresses"
     return webClient.get()
-      .uri("/api/persons/$personId/addresses")
+      .uri(uri)
       .retrieve()
       .bodyToMono(addresses)
+      .onErrorResume {
+          e ->
+        if (!isNotFoundError(e)) {
+          logger.error("get contact address Failed for get request $uri")
+          Mono.error(e)
+        } else {
+          logger.error("get contact address returned NOT_FOUND for get request $uri")
+          Mono.error { PersonNotFoundException("Contacts not found $personId on prison-api") }
+        }
+      }
```

Comment:
we might have to test this scenario to ensure prison API returns a 404 if address not found

Reply:
https://prison-register-dev.hmpps.service.justice.gov.uk/swagger-ui/index.html#/prison-contact-details-resource/getContactDetails 

As mentioned on slack, it appears the swagger does return a 404 for the scenario of a prison missing contact information. We should be fine with this